torch/lib/c10d/reducer.cpp

```diff
@@ -767,6 +769,11 @@ void Reducer::mark_variable_ready(VariableIndex index) {
       }
       // Check that all buckets were completed and had their work kicked off.
       TORCH_INTERNAL_ASSERT(next_bucket_ == buckets_.size());
+      if (static_graph_after_first_iteration() && should_rebuild_buckets()) {
```

Comment:
nit: Add a comment here mentioning why unused parameters need to be pushed at the end so that it's not accidentally changed by developer later on?

Also curious, are there any workflows/use cases where this could actually negatively affect performance?

Reply:
I think this should not result in negative performance, as it is not blocking other all reduces